pkg/execution/state/redis_state/lua/includes/gcra.lua

```diff
@@ -1,103 +1,128 @@
--- performs gcra rate limiting for a given key.
---
--- Returns true on success, false if the key has been rate limited.
-local function gcra(key, now_ms, period_ms, limit, burst, enableThrottleFix)
-	-- Calculate the basic variables for GCRA.
-	local cost = 1                            -- everything counts as a single rqeuest
-
-	local emission  = period_ms / math.max(limit, 1)   -- how frequently we can admit new requests
-	local increment = emission * cost         -- this request's time delta
-  -- BUG: The variance is calculated incorrectly. We should use emission instead of period_ms.
-	local variance  = period_ms * (math.max(burst, 1)) -- variance takes into account bursts
-  if enableThrottleFix then
-    -- NOTE: This fixes the delay variation tolerance calculation
-    variance = emission * (math.max(burst, 1))
-  end
-
-	-- fetch the theoretical arrival time for equally spaced requests
-	-- at exactly the rate limit
+-- applyGCRA runs GCRA
+local function applyGCRA(key, now_ms, period_ms, limit, burst, quantity)
+	---@type { allowed: boolean, limit: integer?, retry_after: integer?, reset_after: integer?, remaining: integer? }
+	local result = {}
```

Comment:
any possible performance concerns returning a table with more fields than the 2 element sequential table before? (maybe memory usage in lua?) 

also looks like the type annotation doesn't capture all the possible keys being set on the table, is it only capturing the ones that matter for actual use?